src/presets.rs

```diff
@@ -159,28 +150,33 @@ pub fn is_safe_prime_with_rng<T: Integer + RandomMod>(
     }
 
     // These are ensured to be odd by the check above.
-    let odd_num = Odd::new(num.clone()).expect("`num` should be odd here");
-    let odd_half_num = Odd::new(num.wrapping_shr_vartime(1)).expect("`num/2` should be odd here");
+    let odd_num = Odd::new(num.clone()).expect("`num` is odd here given the checks above");
+    let odd_half_num = Odd::new(num.wrapping_shr_vartime(1)).expect("The binary rep of `num` ends in `11`, so shifting right by one is guaranteed leave a `1` at the end, so it's odd");
 
-    _is_prime_with_rng(rng, &odd_num) && _is_prime_with_rng(rng, &odd_half_num)
+    _is_prime_with_rng(rng, odd_num) && _is_prime_with_rng(rng, odd_half_num)
 }
 
 /// Checks for primality.
-fn _is_prime_with_rng<T: Integer + RandomMod>(rng: &mut impl CryptoRngCore, num: &Odd<T>) -> bool {
+/// First run a Miller-Rabin test with base 2
+/// If the outcome of M-R is "probably prime", then run a Lucas test
+/// If the Lucas test is inconclusive, run a Miller-Rabin with random base and unless this second
+/// M-R test finds it's composite, then conclude that it's prime.
+fn _is_prime_with_rng<T: Integer + RandomMod>(rng: &mut impl CryptoRngCore, num: Odd<T>) -> bool {
+    let candidate = num.clone();
```

Comment:
Probably a little cleaner would be to call the argument `candidate` right in the method signature.